File: infra/src/main/java/com/beat/infra/persistence/staff/entity/StaffJpaEntity.java
package com.beat.infra.persistence.staff.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity(name = "Staff")
@Table(name = "staff")
public class StaffJpaEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(updatable = false)
	private Long id;

	@Column(nullable = false)
	private String staffName;

	@Column(nullable = false)
	private String staffRole;

	@Column(nullable = false)
	private String staffPhoto;

	@Column(name = "performance_id", nullable = false)
	private Long performanceId;

	protected StaffJpaEntity() {
	}

	private StaffJpaEntity(Long id, String staffName, String staffRole, String staffPhoto, Long performanceId) {
		this.id = id;
		this.staffName = staffName;
		this.staffRole = staffRole;
		this.staffPhoto = staffPhoto;
		this.performanceId = performanceId;
	}

	public static StaffJpaEntity rehydrate(Long id, String staffName, String staffRole, String staffPhoto,
		Long performanceId) {
		return new StaffJpaEntity(id, staffName, staffRole, staffPhoto, performanceId);
	}

	public Long getId() {
		return id;
	}

	public String getStaffName() {
		return staffName;
	}

	public String getStaffRole() {
		return staffRole;
	}

	public String getStaffPhoto() {
		return staffPhoto;
	}

	public Long getPerformanceId() {
		return performanceId;
	}
}

File: infra/src/main/kotlin/com/beat/infra/persistence/staff/entity/StaffJpaEntity.kt
package com.beat.infra.persistence.staff.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity(name = "Staff")
@Table(name = "staff")
class StaffJpaEntity private constructor(
    id: Long?,
    staffName: String,
    staffRole: String,
    staffPhoto: String,
    performanceId: Long,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(updatable = false)
    var id: Long? = id
        protected set

    @Column(nullable = false)
    var staffName: String = staffName
        protected set

    @Column(nullable = false)
    var staffRole: String = staffRole
        protected set

    @Column(nullable = false)
    var staffPhoto: String = staffPhoto
        protected set

    @Column(name = "performance_id", nullable = false)
    var performanceId: Long = performanceId
        protected set

    companion object {
        @JvmStatic
        fun rehydrate(
            id: Long?,
            staffName: String,
            staffRole: String,
            staffPhoto: String,
            performanceId: Long,
        ): StaffJpaEntity = StaffJpaEntity(
            id = id,
            staffName = staffName,
            staffRole = staffRole,
            staffPhoto = staffPhoto,
            performanceId = performanceId,
        )
    }
}
